langchain/sql_database.py

```diff
@@ -110,6 +112,101 @@ def from_uri(
         _engine_args = engine_args or {}
         return cls(create_engine(database_uri, **_engine_args), **kwargs)
 
+    @classmethod
+    def from_databricks(
+        cls,
+        catalog: str,
+        schema: str,
+        host: Optional[str] = None,
+        api_token: Optional[str] = None,
+        warehouse_id: Optional[str] = None,
+        cluster_id: Optional[str] = None,
+        **kwargs: Any,
+    ) -> SQLDatabase:
+        """
+        Class method to create an SQLDatabase instance from a Databricks connection.
+        This method requires the 'databricks-sql-connector' package. If not already
+        installed, it can be added using `pip install databricks-sql-connector`.
+
+        Args:
+            catalog (str): The catalog name in the Databricks database.
+            schema (str): The schema name in the catalog.
+            host (Optional[str]): The host URL of the Databricks instance. If not
+                provided, it will be fetched from the environment variable
+                'DATABRICKS_HOST' or from the current Databricks REPL context.
+                Defaults to None.
+            api_token (Optional[str]): The API token for the Databricks instance.
+                If not provided, it will be fetched from the environment variable
+                'DATABRICKS_API_TOKEN' or from the current Databricks REPL context.
+                Defaults to None.
+            warehouse_id (Optional[str]): The warehouse ID in the Databricks SQL.
+                If provided, the method will configure the connection to use this
+                warehouse. Cannot be used in conjunction with 'cluster_id'. Defaults
+                to None.
+            cluster_id (Optional[str]): The cluster ID in the Databricks Runtime.
+                If provided, the method will configure the connection to use this
+                cluster. Cannot be used in conjunction with 'warehouse_id'. If both
+                'warehouse_id' and 'cluster_id' are None, it will use the cluster ID
+                from the current Databricks REPL context. Defaults to None.
```

Comment:
if running inside a Databricks notebook, it will use the ID of the cluster the notebook is attached to.